lib/blocs/authentication/viewmodels/authentication_bloc.dart

```diff
@@ -23,27 +24,24 @@ class AuthenticationBloc extends Bloc<AuthenticationEvent, AuthenticationState>
       yield state.copyWith(authStatus: AuthStatus.unlocked);
     }
     if (event is OnCreateAccount) {
-      settingsStorage.saveAccount(event.account, event.privateKey);
+      SaveAccountUseCase().run(accountName: event.account, privateKey: event.privateKey);
       // New account --> re-start auth status
       add(const InitAuthStatus());
-      // Set fcm token must be last instruction to allow login, even if there is an error here.
-      await FirebaseMessageTokenRepository().setFirebaseMessageToken(settingsStorage.accountName);
     }
     if (event is OnImportAccount) {
-      settingsStorage.saveAccount(event.account, event.privateKey);
+      SaveAccountUseCase().run(accountName: event.account, privateKey: event.privateKey);
       settingsStorage.privateKeyBackedUp = true;
       // New account --> re-start auth status
       add(const InitAuthStatus());
-      // Set fcm token must be last instruction to allow login, even if there is an error here.
-      await FirebaseMessageTokenRepository().setFirebaseMessageToken(event.account);
     }
     if (event is OnRecoverAccount) {
-      settingsStorage.saveAccount(event.account, event.privateKey);
+      // TODO(RaulUrtecho): - This is redundant - user was already saved when recovery was started.
+      // also a recovery is 1:1 tied to a user account so the account can't change
+      // the key we use as argument here comes from settingsStorage and goes back there.
+      SaveAccountUseCase().run(accountName: event.account, privateKey: event.privateKey);
```

Comment:
So the account and key both are already saved previous reach this event. if so you right no make sense save again here.